output.c

```diff
@@ -356,17 +332,66 @@ output_set_window_title(struct cg_output *output, const char *title)
 static bool
 output_config_apply(struct cg_server *server, struct wlr_output_configuration_v1 *config, bool test_only)
 {
-	struct wlr_output_configuration_head_v1 *head;
+	size_t states_len;
+	struct wlr_output_state *states = wlr_output_configuration_v1_build_state(config, &states_len);
+	if (states == NULL) {
+		return false;
+	}
 
+	for (size_t i = 0; i < states_len; i++) {
+		struct wlr_output_state *state = &states[i];
+
+		bool enabled = state->output->enabled;
+		if (state->committed & WLR_OUTPUT_STATE_ENABLED) {
+			enabled = state->enabled;
+		}
+		if (!enabled) {
+			continue;
+		}
+
+		struct wlr_swapchain *swapchain = NULL;
+		if (!wlr_output_configure_primary_swapchain(state->output, state, &swapchain)) {
```

Comment:
Yeah, that's a good point…

Backend-wide commits are still good to have even without the modifier fallback stuff: they provide atomic output configurations without some half-applied state in case of failures.

I'd prefer to keep the modifier fallback stuff as a follow up of the backend commits, but we will indeed either need to handle this in a helper or in `wlr_backend_commit()`…

Reply:
Agreed.

I for fun tried to hack up a solution in clay (https://git.sr.ht/~kennylevinsen/clay/tree/dc82a8c/item/src/clay/output.c#L236), exposing a specialized version of `wlr_output_configure_primary_swapchain` (`wlr_output_try_for_swapchain`) that just grabs a buffer with the configuration you want, successful or not. 

Seems to work quite well so far, so there's hope.

Reply:
This PR now uses the new helper from [wlroots!4567](https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4567).